envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotsVersions.kt

```diff
@@ -32,18 +36,19 @@ internal class SnapshotsVersions {
         val versionsWithData = versions.compute(group) { _, previous ->
             val version = when (previous) {
                 null -> Version(
-                        clusters = ClustersVersion(newVersion()),
-                        endpoints = EndpointsVersion(newVersion()),
+                        clusters = ClustersVersion(clusters),
+                        endpoints = EndpointsVersion(endpoints),
                         listeners = ListenersVersion(newVersion()),
                         routes = RoutesVersion(newVersion())
                 )
                 else -> {
                     val clustersChanged = previous.clusters != clusters
+                    val haveListenersChanged = previous.listeners != listeners
```

Comment:
could the variables names be consistent? Both with `have` or both without `have`? (I personally prefer shorter version)

Reply:
Shorter it is